### ISSUE TITLE
Allow "user" and "password" to be specified in postage-connections.conf

### ISSUE THREAD

**Issue by stevenwinfield** [workflowproducts/postage#125]
Windows Server 2008 R2
Postage 3.1.3
Postgres 9.6.2

The postage man page explains that lines in postage-connections.conf are of the form:
[server_name]: [standard PostgreSQL connection string]
...but "user" and "password" - which are valid connection string key words - are not accepted, it seems.

The last used username for a given server_name seems to already be stored somewhere, but not the password.

It would be great if "user" and "password" could be taken from lines in postage-connections.conf - assuming the file has the correct permissions set of course (only user readable/writable).

Thanks,
Steve.

**Comment by justintocci** [comment]
You are correct, we did exclude the "user" and "password" keys to discourage users from recording their passwords to disk. 

Once you log in successfully, your user name is stored in a cookie.

Michael, what are our options?

**Comment by justintocci** [comment]
We could also remove the "user" and "password" key exclusion. We originally did that because we were paranoid people would criticise us for poor security practices. At this point, I wonder if that might get us some much needed publicity...

What does the community think? Is it a security risk or do we want to allow "user" and "password" keys in the config file?

**Comment by crosstocci** [comment]
Steven, thanks for the feature request!

We have a couple of options, we could allow you to store the username and password in the configuration file or we could add a "save password" checkbox on the login screen.

If we go with the "save password" checkbox, we'd be storing the password in a cookie.

**Comment by josephtocci** [comment]
Does windows work with .pgpass? This is what pgsql uses, I'm afraid to store the password in a nonstandard place.

**Comment by nunziotocci** [comment]
.pgpass exists on windows as %APPDATA%/postgresql/pgpass.conf

**Comment by stevenwinfield** [comment]
Yes, .pgpass works on windows and this is how pgAdmin3 stored user passwords.

**Comment by justintocci** [comment]
We use the .pgpass on UNIX (I just tested it on Mac) 

We also use the pgpass.conf on Windows. (I just tested it on Windows) The actual path that I had to use was C:\Users\jtocci\AppData\Roaming\postgresql\pgpass.conf. In my postage-connections.conf I was using a host name and in the pgpass.conf I had a different host name that resolves to the same place, but the mis-match prevented it from working. Once I made them the same it worked. 

https://www.postgresql.org/docs/current/static/libpq-pgpass.html

Now I am wondering what we need to do. I like the pgpass solution because it's standard. As of right now, I'm thinking we should just add a convenience button to create/edit the pgpass file (like we have for postage-connections.conf) and call it a day. @stevenwinfield please feel free to weigh in on this.

**Comment by stevenwinfield** [comment]
I think using pgpass, and allowing it to be edited, would be great - it would allow users to use wildcards too. It would be handy if something were nearby to remind you of pgpass's fileformat - I can never quite remember the field order.

Thanks for looking at this!

**Comment by justintocci** [comment]
Nunzio:

1) now that I'm using the PGPASS on Windows, I'm noticing it doesn't always work the first time I try it. Steps to reproduce: open postage, choose a server that has a valid pgpass line, click Log In. This error pops up:
![screen shot 2017-03-02 at 2 40 06 pm](https://cloud.githubusercontent.com/assets/14250339/23526061/4f195de2-ff56-11e6-8133-1aa813eb5c3a.JPG)
Then if I click "ok" on the error and click "Log In" a second time it lets me in.

2) Let's put together an edit window like postage-connections.conf, however, lets add a "help" button that sends you to the documentation page. Also, if it's easy to do, I tested the PGPASS with "#" comment lines and it worked fine. If you have to create the PGPASS file, it would be cool to have a few comments in there explaining the format.

**Comment by nunziotocci** [comment]
@justintocci That looks like your VPN wasn't connected or something. I'm almost certain that it has nothing to do with pgpass.conf.
